atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/metrics/KeyValueSnapshotEventRecorder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api.metrics;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+/**
+ * Updates metrics for work done as part of reading a transactional snapshot from the key-value-service.
+ */
+public interface KeyValueSnapshotEventRecorder {
```

Comment:
I'd probably name this as MetricRecorder, MetricEmitter, or something along these lines. EventRecorder to me sounds like a class that is an intermediary holder of Events that will then pass them to a third class for something to be done with such events.

Reply:
Makes sense, will rename (I based this off `LeadershipEvents`, but you're right that the naming is strange).